crates/astria-composer/src/searcher.rs

```diff
@@ -1,358 +1,72 @@
-        Ok(Self {
-            inner,
-        })
-    }
-
-    /// Wrapper around [`Client::abci_info`] with a 1s timeout.
-    async fn abci_info(self) -> eyre::Result<abci::response::Info> {
-        use sequencer_client::Client as _;
-        tokio::time::timeout(Duration::from_secs(1), self.inner.abci_info())
-            .await
-            .wrap_err("request timed out")?
-            .wrap_err("RPC returned with error")
-    }
+pub struct Searcher {
+    status: watch::Sender<SearcherStatus>,
 }
 
 impl Searcher {
-    /// Constructs a new Searcher service from config.
-    ///
-    /// # Errors
-    ///
-    /// Errors are returned in the following scenarios:
-    /// + failed to connect to the eth RPC server;
-    /// + failed to construct a sequencer clinet
-    pub(super) async fn from_config(cfg: &Config) -> eyre::Result<Self> {
-        // connect to eth node
-        let eth_client = EthClient::connect(&cfg.execution_url)
-            .await
-            .wrap_err("failed connecting to eth")?;
-
-        // connect to sequencer node
-        let sequencer_client = SequencerClient::new(&cfg.sequencer_url)
-            .wrap_err("failed constructing sequencer client")?;
-
-        let rollup_chain_id = cfg.chain_id.clone();
-        let (status, _) = watch::channel(Status::default());
-
-        Ok(Searcher {
-            eth_client,
-            sequencer_client,
-            rollup_chain_id,
-            status,
-            conversion_tasks: JoinSet::new(),
-            submission_tasks: JoinSet::new(),
-        })
-    }
-
-    pub(crate) fn subscribe_to_state(&self) -> watch::Receiver<Status> {
+    pub(crate) fn subscribe_to_state(&self) -> watch::Receiver<SearcherStatus> {
         self.status.subscribe()
     }
 
-    /// Serializes and signs a sequencer tx from a rollup tx.
-    fn handle_pending_tx(&mut self, rollup_tx: Transaction) {
-        let chain_id = self.rollup_chain_id.clone();
-
-        self.conversion_tasks.spawn_blocking(move || {
-            // FIXME(https://github.com/astriaorg/astria/issues/215): need to set and track
-            // nonces for an actual fixed funded key/account.
-            // For now, each transaction is transmitted from a new account with nonce 0
-            let sequencer_key = ed25519_consensus::SigningKey::new(rand::thread_rng());
-            let nonce = Nonce::from(0);
-
-            // Pack into sequencer tx
-            let data = rollup_tx.rlp().to_vec();
-            let chain_id = chain_id.into_bytes();
-            let seq_action = SequencerAction::SequenceAction(SequenceAction::new(chain_id, data));
-            let unsigned_tx = UnsignedSequencerTx::new_with_actions(nonce, vec![seq_action]);
+    // FIXME: How do we start up multiple providers if we are reading from Config?
+    async fn setup_providers_from_config(
+        cfg: &Config,
+    ) -> Result<Vec<(Box<dyn StreamingClient<Error = eyre::Error>>, RollupChainId)>, eyre::Error>
```

Comment:
i dont think we need to be generic over providers at this point. this should be a later change.

Reply:
I started out this PR with the intent of adding multiple providers, i.e. #214 after talking to @joroshiba and this is the only real new functionality added to the crate with this PR. But we can split it up if that is easier. Would defer to what others think